src/main/java/bio/terra/workspace/service/resource/WsmResourceType.java

```diff
@@ -50,6 +68,25 @@ public static WsmResourceType fromSql(String dbString) {
         "Deeserialization failed: no matching resource type for " + dbString);
   }
 
+  /**
+   * Convert from api type to WsmResourceType. In some contexts, the API input can be null, so if
+   * the input is null we return null and leave it to the caller to raise any error.
```

Comment:
"the null returns null, but invalid throws" is a little subtle to me. Do we really need 2 forms of "could not convert?" Can the clients do their own null checking?